mesonbuild/interpreter.py

```diff
@@ -2501,19 +2497,14 @@ def emit_base_options_warnings(self, enabled_opts):
 
     def add_base_options(self, compiler):
         enabled_opts = []
-        proj_opt = self.environment.cmd_line_options.projectoptions
         for optname in compiler.base_options:
             if optname in self.coredata.base_options:
                 continue
             oobj = compilers.base_options[optname]
-            for po in proj_opt:
-                if po.startswith(optname + '='):
-                    opt, value = po.split('=', 1)
-                    oobj.set_value(value)
-                    if oobj.value:
-                        enabled_opts.append(opt)
-                    break
-            self.coredata.base_options[optname] = oobj
+            if optname in self.environment.cmd_line_options:
+                oobj.set_value(self.environment.cmd_line_options[optname])
+                enabled_opts.append(optname)
+            self.coredata. base_options[optname] = oobj
```

Comment:
TIL: the `.` operator ignores space.

Reply:
oh... didn't know that. Just a typo, let's fix that after-merge, I don't want to wait another 5h to run the CI on trivial change like that. appveyor should be done in a few minutes.

Reply:
I think we can leave it in as an easter egg for people to find, check, and learn 😉